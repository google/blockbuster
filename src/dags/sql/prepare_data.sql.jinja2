{% set days_since_lead_sql = 'DATETIME_DIFF(effectiveDate, EXTRACT(DATETIME FROM
                              TIMESTAMP_SECONDS(CAST(RECENT_most_recent_lead AS INT64))), DAY)' %}
WITH Buckets AS (
  SELECT
    BB_id,
    CASE FLOOR(RAND()*10) WHEN 0 THEN 'TEST' WHEN 1 THEN 'VALIDATE' ELSE 'TRAIN' END AS MLDataSplit
  FROM (
    SELECT
      DISTINCT RECENT_BB_id AS BB_id
    FROM `{{ features_table }}`
    WHERE RECENT_most_recent_lead <> ''
      AND {{ days_since_lead_sql }} <= 7
  )
)
SELECT
  * EXCEPT({{ exclude_from_output|join(', ') }}),
  {{ days_since_lead_sql }} AS days_since_recent_lead
FROM `{{ features_table }}` AS Features
LEFT OUTER JOIN
  Buckets ON Features.RECENT_BB_id = Buckets.BB_id
WHERE RECENT_most_recent_lead <> ''
  AND {{ days_since_lead_sql }} <= {{ inclusion_recency_days }};
