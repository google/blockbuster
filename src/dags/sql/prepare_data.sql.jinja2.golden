
WITH Buckets AS (
  SELECT
    BB_id,
    CASE FLOOR(RAND()*10) WHEN 0 THEN 'TEST' WHEN 1 THEN 'VALIDATE' ELSE 'TRAIN' END AS MLDataSplit
  FROM (
    SELECT
      DISTINCT RECENT_BB_id AS BB_id
    FROM `project.dataset.table`
    WHERE RECENT_most_recent_lead <> ''
      AND DATETIME_DIFF(effectiveDate, EXTRACT(DATETIME FROM
                              TIMESTAMP_SECONDS(CAST(RECENT_most_recent_lead AS INT64))), DAY) <= 7
  )
)
SELECT
  * EXCEPT(col1, col2, col3),
  DATETIME_DIFF(effectiveDate, EXTRACT(DATETIME FROM
                              TIMESTAMP_SECONDS(CAST(RECENT_most_recent_lead AS INT64))), DAY) AS days_since_recent_lead
FROM `project.dataset.table` AS Features
LEFT OUTER JOIN
  Buckets ON Features.RECENT_BB_id = Buckets.BB_id
WHERE RECENT_most_recent_lead <> ''
  AND DATETIME_DIFF(effectiveDate, EXTRACT(DATETIME FROM
                              TIMESTAMP_SECONDS(CAST(RECENT_most_recent_lead AS INT64))), DAY) <= 7;
